arangod/Cluster/MaintenanceFeature.h

```diff
@@ -384,6 +400,13 @@ class MaintenanceFeature : public application_features::ApplicationFeature {
                      std::shared_ptr<VPackBuffer<uint8_t>>> _dbErrors;
 
   
+  /// @brief lock for shard version map
+  mutable arangodb::Mutex _versionLock;
```

Comment:
You could use `std::mutex` :)

Reply:
will not change for this lock only. should be overhauled in other pr.